src/source/JavaGenerator.scala

```diff
@@ -140,7 +140,10 @@ class JavaGenerator(spec: Spec) extends Generator(spec) {
       writeDoc(w, doc)
 
       javaAnnotationHeader.foreach(w.wl)
-      w.w(s"${javaClassAccessModifierString}abstract class $javaClass$typeParamList").braced {
+      val isAbstractClass = (spec.javaGenerateInterfaces == false) || i.ext.cpp || i.methods.exists(_.static) || i.consts.size > 0
```

Comment:
> Just make the CppProxy constructors private and that should do. (mknejp)

I believe that the constructors he's referring to are already private. Does this mean that the "i.ext.cpp" check can be removed?

> Also, a Java interface can have static final member variables . . . (artwyman)

Are you saying that the "i.consts.size > 0" check can be removed?